experimental/traffic-portal/src/app/core/statuses/statuses-table/statuses-table.component.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Component, OnInit } from "@angular/core";
+import { BehaviorSubject } from "rxjs";
+import { ResponseStatus } from "trafficops-types";
+
+import { ServerService } from "src/app/api";
+import { ContextMenuItem } from "src/app/shared/generic-table/generic-table.component";
+import { NavigationService } from "src/app/shared/navigation/navigation.service";
+
+/**
+ * StatusesTableComponent is the controller for the statuses page - which
+ * principally contains a table.
+ */
+@Component({
+	selector: "tp-statuses-table",
+	styleUrls: ["./statuses-table.component.scss"],
+	templateUrl: "./statuses-table.component.html",
+})
+export class StatusesTableComponent implements OnInit {
+
+	/** All of the statues which should appear in the table. */
+	public statuses: ResponseStatus[] = [];
+
+	/** Definitions of the table's columns according to the ag-grid API */
+	public columnDefs = [
+		{
+			field: "name",
+			headerName: "Name",
+			hide: false
+		},
+		{
+			field: "description",
+			headerName: "Description",
+			hide: false
+		}];
+
+	/** The current search text. */
+	public searchText = "";
+
+	/** Definitions for the context menu items (which act on statuses data). */
+	public contextMenuItems: Array<ContextMenuItem<ResponseStatus>> = [
+		{
+			href: (u: ResponseStatus): string => `${u.id}`,
+			name: "View Status Details"
+		},
+		{
+			href: (): string => "new",
```

Comment:
`href` is allowed to just be a string if it's a constant value

Reply:
removed and added fab button to add new status as followed in other pages